web/modules/custom/dpl_library_agency/src/GeneralSettings.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\dpl_library_agency;
+
+use Drupal\dpl_react\DplReactConfigBase;
+
+/**
+ * Class that handles general settings.
+ */
+class GeneralSettings extends DplReactConfigBase {
+
+  const EXPIRATION_WARNING_DAYS_BEFORE_CONFIG = 6;
+  const RESERVATION_DETAIL_ALLOW_REMOVE_READY_RESERVATIONS = FALSE;
+  const INTEREST_PERIODS_CONFIG = '180-6 months';
+  const DEFAULT_INTEREST_PERIOD_CONFIG = [
+    "value" => "180",
+    "label" => "6 months",
+  ];
+  const RESERVATION_SMS_NOTIFICATIONS_ENABLED = TRUE;
+  const PAUSE_RESERVATION_INFO_URL = '';
+  const REDIRECT_ON_BLOCKED_URL = '';
+  const BLOCKED_PATRON_E_LINK_URL = '';
+  const EREOLEN_MY_PAGE_URL = '';
+  const PAUSE_RESERVATION_START_DATE_CONFIG = '';
+
+  /**
+   * Gets the configuration key for general settings.
+   */
+  public function getConfigKey(): string {
+    return 'dpl_library_agency.general_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfig(): array {
+    return $this->legacyConfig();
+  }
+
+  /**
+   * Get the interestPeriodConfiguration.
+   *
+   * @return array[]
+   *   Array containing the collected interestPeriodConfiguration.
+   */
+  public function getInterestPeriodsConfig(): array {
+    $interestPeriods = self::getInterestPeriodsAsArray() ?? self::INTEREST_PERIODS_CONFIG;
+    $defaultInterestPeriod = self::getDefaultInterestPeriodAsArray($interestPeriods) ?? self::DEFAULT_INTEREST_PERIOD_CONFIG;
+
+    $interestPeriodsConfig['interestPeriods'] = [];
+    foreach ($interestPeriods as $key => $value) {
+      $interestPeriodsConfig['interestPeriods'][] = [
+        'value' => $key,
+        'label' => $value,
+      ];
+    }
+
+    $interestPeriodsConfig['defaultInterestPeriod'] = $defaultInterestPeriod;
+
+    return $interestPeriodsConfig;
+  }
```

Comment:
Throughout this file you use camelCase for variable names. According to drupal coding style they should be lowercase. Isn't phpcs picking this up?

Reply:
Updated this. Please see latest commit.

You are right, this is my fault. 
Phpcs is not picking this up on my local dev. I have to look into this.

Reply:
👍